GirafRest/Controllers/ActivityController.cs

```diff
@@ -0,0 +1,112 @@
+        /// UserNotFound, NotAuthorized, WeekNotFound or InvalidDay.</returns>
+        [HttpPost("{userId}/{weekplanName}/{weekYear}/{weekNumber}/{weekDay}")]
+        [Authorize]
+        public async Task<Response<ActivityDTO>> PostActivity([FromBody] ActivityDTO newActivity, string userId, string weekplanName, int weekYear, int weekNumber, Days weekDay)
+        {
+            if (newActivity == null)
+                return new ErrorResponse<ActivityDTO>(ErrorCode.MissingProperties);
+
+            GirafUser user = await _giraf.LoadUserWithWeekSchedules(userId);
+            if (user == null)
+                return new ErrorResponse<ActivityDTO>(ErrorCode.UserNotFound);
+
+            // check access rights
+            if (!(await _authentication.HasEditOrReadUserAccess(await _giraf._userManager.GetUserAsync(HttpContext.User), user)))
+                return new ErrorResponse<ActivityDTO>(errorCode: ErrorCode.NotAuthorized);
+
+            var dbWeek = user.WeekSchedule.FirstOrDefault(w => w.WeekYear == weekYear && w.WeekNumber == weekNumber && string.Equals(w.Name, weekplanName));
+            if (dbWeek == null)
+                return new ErrorResponse<ActivityDTO>(errorCode: ErrorCode.WeekNotFound);
+
+            Weekday dbWeekDay = dbWeek.Weekdays.FirstOrDefault(day => day.Day == weekDay);
+            if (dbWeekDay == null)
+                return new ErrorResponse<ActivityDTO>(ErrorCode.InvalidDay);
+
+            int order = dbWeekDay.Activities.Select(act => act.Order).DefaultIfEmpty(0).Max();
+            order++;
+
+            Activity dbActivity = new Activity(dbWeekDay, new Pictogram() { Id = newActivity.Pictogram.Id }, order, ActivityState.Normal);
+            _giraf._context.Activities.Add(dbActivity);
+            await _giraf._context.SaveChangesAsync();
+
+            return new Response<ActivityDTO>(new ActivityDTO(dbActivity));
+        }
+
+        /// <summary>
+        /// Delete an activity with a given id.
+        /// </summary>
+        /// <param name="userId">id of the user you want to delete an activity for.</param>
+        /// <param name="activityId">id of the activity you want to delete.</param>
+        /// <returns>Returns success response else UserNotFound, NotAuthorized or ActivityNotFound.</returns>
+        [HttpDelete("{userId}/delete/{activityId}")]
+        [Authorize]
+        public async Task<Response> DeleteActivity(string userId, long activityId)
+        {
+            GirafUser user = await _giraf.LoadUserWithWeekSchedules(userId);
+            if (user == null)
+                return new ErrorResponse(ErrorCode.UserNotFound);
+
+            // check access rights
+            if (!(await _authentication.HasEditOrReadUserAccess(await _giraf._userManager.GetUserAsync(HttpContext.User), user)))
+                return new ErrorResponse(errorCode: ErrorCode.NotAuthorized);
```

Comment:
consistency with ErrorResponse, line 88, 92 and 96? 😃

Reply:
ErrorResponse syntax is now consistent 🐤